grpc/execution/server.go

```diff
@@ -92,6 +92,11 @@ func (s *ExecutionServiceServer) DoBlock(ctx context.Context, req *executionv1.D
 		return nil, fmt.Errorf("failed to insert block into blockchain (n=%d)", n)
 	}
 
+	// remove txs from original mempool
+	for _, tx := range block.Transactions() {
```

Comment:
Should we be removing all transactions in the request even if they don't make the block, or maybe doing something with transactions that don't make it into the block?

Reply:
all the transactions in here did make it into the block, so this should be fine

Reply:
We have filtering to stop transactions that can't be executed from being included. Those txs will NEVER be in a block, should they remain in the mempool?